extension/wallet/src/protocol/make_create_swap_payload.rs

```diff
@@ -23,54 +36,76 @@ pub async fn make_buy_create_swap_payload(
         *guard
     };
 
-    make_create_swap_payload(
-        name,
-        current_wallet,
-        sell_amount,
-        usdt_asset_id,
-        btc_asset_id,
-    )
-    .await
+    make_create_swap_payload(wallet_name, sell_amount, usdt_asset_id, btc_asset_id).await
 }
 
+/// Constructs a new [`CreateSwapPayload`] with the given Bitcoin amount.
+///
+/// This will select UTXOs from the wallet to cover the given amount.
+#[wasm_bindgen]
 pub async fn make_sell_create_swap_payload(
-    name: String,
-    current_wallet: &Mutex<Option<Wallet>>,
+    wallet_name: String,
+    btc: String,
+) -> Result<JsValue, JsValue> {
+    let btc = map_err_from_anyhow!(Amount::from_str_in(&btc, Denomination::Bitcoin))?;
+
+    let payload = map_err_from_anyhow!(make_sell_create_swap_payload_w(wallet_name, btc).await)?;
+    let payload = map_err_from_anyhow!(JsValue::from_serde(&payload))?;
+
+    Ok(payload)
+}
+
+async fn make_sell_create_swap_payload_w(
+    wallet_name: String,
     sell_amount: Amount,
 ) -> Result<CreateSwapPayload, Error> {
     let btc_asset_id = {
         let guard = BTC_ASSET_ID.lock().expect_throw("can get lock");
         *guard
     };
-    make_create_swap_payload(
-        name,
-        current_wallet,
-        sell_amount,
-        btc_asset_id,
-        btc_asset_id,
-    )
-    .await
+
+    make_create_swap_payload(wallet_name, sell_amount, btc_asset_id, btc_asset_id).await
 }
 
 async fn make_create_swap_payload(
-    name: String,
-    current_wallet: &Mutex<Option<Wallet>>,
+    wallet_name: String,
     sell_amount: Amount,
     sell_asset: AssetId,
     fee_asset: AssetId,
 ) -> Result<CreateSwapPayload, Error> {
-    let wallet = current(&name, current_wallet)
-        .await
-        .map_err(Error::LoadWallet)?;
-    let blinding_key = wallet.blinding_key();
-
-    let utxos = get_txouts(&wallet, |utxo, txout| {
-        Ok({
-            let unblinded_txout = txout.unblind(SECP256K1, blinding_key)?;
+    let client = {
+        let guard = ESPLORA_API_URL.lock().expect_throw("can get lock");
+        EsploraClient::new(guard.clone())
+    };
+
+    let (utxos, secret_key, address, blinding_key) = {
+        let mut wallet = current(&wallet_name, &LOADED_WALLET)
+            .await
+            .map_err(Error::LoadWallet)?;
+        wallet.sync(&client).await.map_err(Error::SyncWallet)?;
+        (
+            wallet.utxos(),
```

Comment:
Hmm, what is the point of calling `sync` if the only thing we do is take the UTXOs out again? :sweat_smile: 

If you are unsure on how much functionality should move into the wallet here, then just do this:

```rust
let utxos = client.get_utxos(wallet.address()).await?;
```

Although ideally, we can add a better API to `Wallet` that reduces the amount of code we have to write in here.

What we are doing in here is coin-selection. Can we have a `Wallet::coin_select` API that returns a list of UTXOs that is enough to fund a certain amount?

Reply:
yep agreed, I think coin select should be pretty easy to move to the wallet